modules/network/src/main/java/org/apache/ignite/internal/network/serialization/FieldDescriptor.java

```diff
@@ -27,38 +27,50 @@
     /**
      * Name of the field.
      */
-    @NotNull
     private final String name;
 
     /**
      * Type of the field.
      */
-    @NotNull
     private final Class<?> clazz;
 
     /**
      * Field type's descriptor id.
      */
     private final int typeDescriptorId;
 
+    /**
+     * The class in which the field is declared.
+     */
+    private final Class<?> declaringClass;
```

Comment:
Why do we need it?

Reply:
Discussed in a private chat: this will be solve in an upcoming PR for IGNITE-16165 (coming in 1-2 days).